### PR TITLE
【PaddlePaddle Hackathon 4】add paddle index_select op

### PR DESCRIPTION
**Pull request by AndPuQing**
### Details:

add paddle index_select op

### reference

https://docs.openvino.ai/latest/openvino_docs_ops_movement_Gather_8.html

https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/index_select_en.html#index-select

image

### CODE REVIEW

File: src/frontends/paddle/tests/test_models/gen_scripts/generate_index_select.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        )
+
+    return outs[0]
+
+
+def main():
+    x = np.random.rand(8, 24, 32).astype(data_type)
+    index = np.random.randint(0, 7, (5)).astype("int32")
+    index_select("index_select_axis_0", x, index, axis=0)
+    index_select("index_select_axis_1", x, index, axis=1)
```
Comment (OpenVINO-dev-contest) [inline src/frontends/paddle/tests/test_models/gen_scripts/generate_index_select.py:54]: hi @AndPuQing Can axis be negative here ?

Comment (AndPuQing) [reply src/frontends/paddle/tests/test_models/gen_scripts/generate_index_select.py:54]: image
Yes, it supports it. I added a unit test for this

File: src/frontends/paddle/src/op/index_select.cpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace ov {
+namespace frontend {
+namespace paddle {
+namespace op {
+NamedOutputs index_select(const NodeContext& node) {
+    auto data = node.get_input("X");
+    auto index = node.get_input("Index");
+    Output<Node> axis_node;
+    const auto axis_value = node.get_attribute<int>("dim");
```
Comment (OpenVINO-dev-contest) [inline src/frontends/paddle/src/op/index_select.cpp:16]: can we put a default number for ```dim``` here ？

**Review by OpenVINO-dev-contest — APPROVED** [review]

**Comment by yuxu42** [comment]
@meiyang-intel Could you please review the PR?

**Comment by xczhai** [comment]
could you help update the doc?
https://github.com/openvinotoolkit/openvino/blob/master/docs/MO_DG/prepare_model/Supported_Frameworks_Layers.md

**Comment by AndPuQing** [comment]
> could you help update the doc? https://github.com/openvinotoolkit/openvino/blob/master/docs/MO_DG/prepare_model/Supported_Frameworks_Layers.md

done.

**Review by ceciliapeng2011 — APPROVED** [review]

**Pull request merged**